utility_analysis_new/histograms.py

```diff
@@ -152,31 +162,36 @@ def bins_to_histogram(bins):
 
 
 def _list_to_contribution_histograms(
-        histograms: List[Histogram]) -> ContributionHistograms:
+        histograms: List[Histogram]) -> DatasetHistograms:
     """Packs histograms from a list to ContributionHistograms."""
     for histogram in histograms:
-        if histogram.name == "CrossPartitionHistogram":
-            cross_partition_histogram = histogram
-        else:
-            per_partition_histogram = histogram
-    return ContributionHistograms(cross_partition_histogram,
-                                  per_partition_histogram)
-
-
-def _compute_cross_partition_histogram(
+        if histogram.name == HistogramType.L0_CONTRIBUTIONS:
+            l0_contributions = histogram
+        elif histogram.name == HistogramType.LINF_CONTRIBUTIONS:
+            linf_contributions = histogram
+        elif histogram.name == HistogramType.COUNT_PER_PARTITION:
+            count_per_partition = histogram
+        elif histogram.name == HistogramType.COUNT_PRIVACY_ID_PER_PARTITION:
+            privacy_id_per_partition_count = histogram
+    return DatasetHistograms(l0_contributions, linf_contributions,
```

Comment:
This assumes all of (lo_contributions, linf_contributions, count_per_partition, privacy_id_per_partition_count) will be not None, and so is brittle. Maybe we could refactor a little to not assume this or assert that all of them are not None?

Reply:
Yeah, good point! In principle it's fine that some histograms are `None`, but we shouldn't crash :). I've initialized all variable to None and update the test.